### DISCUSSION TITLE
Zephyr Profiling

### DISCUSSION THREAD

**Discussion by svetelna** [zephyrproject-rtos/zephyr#33536] [Q&A] [answered]
Hello,
I would like to make a profiling of a Zephyr application.
So far, I have found the "Footprint and Memory Usage" static analysis [1], "Timing functions" for measuring the execution time of a code section [2], and "Thread Analyzer" [3] for getting stack and CPU runtime usage of the threads.

However, I did not found out how to collect heap info (heap usage) during runtime and CPU time spent on specific tasks.
Is there any possibility of collecting this information?

[1] https://docs.zephyrproject.org/latest/guides/optimizations/tools.html
[2] https://docs.zephyrproject.org/latest/reference/timing_functions/index.html
[3] https://docs.zephyrproject.org/latest/guides/flash_debug/thread-analyzer.html

**Comment by carlescufi** [comment, marked as answer]
I think you could use `sys_heap_print_info`: https://github.com/zephyrproject-rtos/zephyr/blob/master/lib/os/heap-validate.c#L386

**Reply by svetelna** [reply]
Thanks for reply.
